### PR TITLE
Make service compatible with Devstack

### PR DESCRIPTION
**Pull request by kdmccormick**
**Ticket:** https://openedx.atlassian.net/browse/EDUCATOR-4017
**Team:** @edx/educator-neem 

This PR makes changes (most notably, adding `settings/devstack.py`) allowing the Registrar service to be compatible with Devstack. 

[This Devstack PR](https://github.com/edx/devstack/pull/387) will actually add Registrar as an optional service in Devstack.

### CODE REVIEW

File: registrar/settings/devstack.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+)
+
+# OAuth2 variables specific to backend service API calls.
+BACKEND_SERVICE_EDX_OAUTH2_KEY = os.environ.get('BACKEND_SERVICE_EDX_OAUTH2_KEY', 'registrar-backend-service-key')
+BACKEND_SERVICE_EDX_OAUTH2_SECRET = os.environ.get('BACKEND_SERVICE_EDX_OAUTH2_SECRET', 'registrar-backend-service-secret')
+BACKEND_SERVICE_EDX_OAUTH2_PROVIDER_URL = os.environ.get(
+    'BACKEND_SERVICE_EDX_OAUTH2_PROVIDER_URL', 'http://edx.devstack.lms:18000/oauth2',
+)
+
+#JWT_AUTH['JWT_ISSUERS'].append({
```
Comment (kdmccormick) [inline registrar/settings/devstack.py:38]: We will definitely eventually want this commented-out code, but I wasn't sure how to configure `JWT_ISSUERS` right now.

File: registrar/settings/utils.py [hunk 2]
```diff
@@ -1,12 +1,128 @@
-from os import environ
+import platform
+import sys
+from logging.handlers import SysLogHandler
+from os import environ, path
 
 from django.core.exceptions import ImproperlyConfigured
 
 
+# Duplicated from github.com/edx/credentials
```
Comment (schenedx) [inline registrar/settings/utils.py]: No need for this comment. The changes here will deviate as time progress

File: registrar/settings/devstack.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        'ATOMIC_REQUESTS': False,
+        'CONN_MAX_AGE': 60,
+    }
+}
+
+STATICFILES_STORAGE = os.environ.get('STATICFILES_STORAGE', 'django.contrib.staticfiles.storage.StaticFilesStorage')
+STATIC_URL = os.environ.get('STATIC_URL', '/static/')
+
+# OAuth2 variables specific to social-auth/SSO login use case.
+SOCIAL_AUTH_EDX_OAUTH2_KEY = os.environ.get('SOCIAL_AUTH_EDX_OAUTH2_KEY', 'registrar-sso-key')
```
Comment (schenedx) [inline registrar/settings/devstack.py]: If these don't work on devstack, do not have them here either. Once we figured out how to do this, we will add them in later.

**Review by schenedx — APPROVED** [review]
Just one minor comment. LGTM 👍

File: registrar/settings/devstack.py (hunk 3 above)
Comment (kdmccormick) [reply registrar/settings/devstack.py]: :+1:

File: registrar/settings/utils.py (hunk 2 above)
Comment (kdmccormick) [reply registrar/settings/utils.py]: :+1:

**Pull request merged**